### PR TITLE
Revert "Remove `usePagination` and `SharedText` from exports"

### PR DESCRIPTION
**Pull request by ericclemmons**
Reverts aws-amplify/amplify-ui#774

### CODE REVIEW

**Review by zchenwei — APPROVED** [review]
LGTM

File: packages/react/__tests__/exports.ts [hunk 1]
```diff
@@ -1373,6 +1373,7 @@ describe('@aws-amplify/ui-react', () => {
           "ScrollView",
           "SearchField",
           "SelectField",
+          "SharedText",
```
Comment (ericclemmons) [inline packages/react/__tests__/exports.ts:1376]: @reesscot This should still be removed, right?

Comment (zchenwei) [reply packages/react/__tests__/exports.ts:1376]: yea, that should be removed for sure

Comment (reesscot) [reply packages/react/__tests__/exports.ts:1376]: Yup, I can add the removal back in followup pr

**Review by slaymance — APPROVED** [review]
Verified locally that this fixes the docs build issue.

**Pull request closed without merging**